Security/Web_Security.mediawiki

```diff
@@ -0,0 +1,614 @@
+* [https://w3c.github.io/webappsec-subresource-integrity/ Subresource Integrity W3C Standard]
+
+
+= X-Content-Type-Options =
+
+<tt>X-Content-Type-Options</tt> is a header supported by Internet Explorer and Chrome that tells it not to load scripts and stylesheets unless the server indicates the correct MIME type. Without this header, these browsers can incorrectly detect files as scripts and stylesheets, leading to XSS attacks. As such, all sites must set the <tt>X-Content-Type-Options</tt> header and the appropriate MIME types for files that they serve.
+
+== Examples ==
+
+<pre># Prevent IE and Chrome from incorrectly detecting non-scripts as scripts
+X-Content-Type-Options: nosniff</pre>
+
+== See Also ==
+
+* [https://msdn.microsoft.com/en-us/library/gg622941%28v=vs.85%29.aspx Microsoft on Reducing MIME Type Security Risks]
+
+
+= X-Frame-Options =
+
+<tt>X-Frame-Options</tt> is an HTTP header that allows sites control over how your site may be framed within an iframe. Clickjacking is a practical attack that allows malicious sites to trick users into clicking links on your site even though they may appear to be something else entirely. As such, the use of the X-Frame-Options header is mandatory for all new websites, and all existing websites are expected to add support for X-Frame-Options as soon as possible.
+
+Sites that require the ability to be iframed must either use the <tt>ALLOW-FROM</tt> directive or employ JavaScript defenses to prevent clickjacking from malicious origins.
+
+== Directives ==
+
+* <tt>DENY</tt>: disallow allow attempts to iframe site (recommended)
+* <tt>SAMEORIGIN</tt>: allow the site to iframe itself
+* <tt>ALLOW-FROM <em>uri</em></tt>: allow <em>uri</em> to iframe site (not supported in Chrome and Safari)
+
+== Examples ==
+
+<pre># Block site from being iframed
+X-Frame-Options: DENY</pre>
+
+<pre># Only allow my site to frame itself
+X-Frame-Options: SAMEORIGIN</pre>
+
+== See Also ==
+
+* [https://developer.mozilla.org/en-US/docs/Web/HTTP/X-Frame-Options MDN on X-Frame-Options]
+* [https://www.owasp.org/index.php/Clickjacking_Defense_Cheat_Sheet OWASP Clickjacking Defense Cheat Sheet]
+
+
+= X-XSS-Protection =
+
+<tt>X-XSS-Protection</tt> is a feature of Internet Explorer and Chrome that stops pages from loading when they detect reflected cross-site scripting (XSS) attacks. New sites should use this header, but it is only recommended for existing sites, given the small but possible risk of false positives.
+
+== Examples ==
+
+<pre># Block pages from loading when they detect reflected XSS attacks
+X-XSS-Protection: 1; mode=block</pre>
+</div>
+
+
+= Version History =
+
+{| class="wikitable" style="width: 100%;"
+|-
+! scope="col" style="width: 6em;" | Version
+! scope="col" style="width: 6em;" | Editor
+! Changes
+|-
+| align="center" | 1.0
+| align="center" | April
+| Initial document creation
+|}
+
+
+
+= Web Security Cheat Sheet =
```

Comment:
l490 I would argue that this should be at the top of the document with a little summary along the lines of:

This is the web security document. DIfferent websites require different security headers, settings, encryption, etc.
The table below will help you figure out which features your web site should implement.